comp/core/workloadmeta/types.go

```diff
@@ -677,14 +681,25 @@ func (n KubernetesNode) String(verbose bool) string {
 
 var _ Entity = &KubernetesNode{}
 
+// Languages represents languages detected for containers of a pod
+type Languages struct {
+	ContainerLanguages     map[string][]languagemodels.Language
+	InitContainerLanguages map[string][]languagemodels.Language
+}
+
+// LanguageDetection defines the state of language detection for a given deployment
```

Comment:
I think we could improve the naming to distinguish between:
- Languages detected by node agents
- Languages found in the deployment resource from kubernetes

We should also add some context as code comment. Also, I assume that we would need to update the `merge` function in workloadmeta. It is used when some object can be updated by two concurrent sources.

Reply:
Agree for the naming.

Regarding updating merge, you `merge` entities that get collected by different sources. 
In our case,  the sources are `kubeapiserver collector` and `languagedetection` (the latter as an external source). The entity fields updated by these two sources are completely disjoint normally as `languagedetection` should update only languages reported by node agents. This being said, I think it is going to work even without updating the `merge` method of the deployment entity. 

HOWEVER, I also think it is still good to update the merge method in order to rely on the programmer to not mix updated fields by the two sources. 

WDYT

Reply:
IMO we can keep the original method as long as we cover the possible cases with unit tests.